holoviews/operation/datashader.py

```diff
@@ -448,6 +448,12 @@ def _apply_datashader(self, dfdata, cvs_fn, agg_fn, agg_kwargs, x, y):
                     val = val.astype(np.float64)
                     val[neg1] = np.nan
                 agg[col] = ((y.name, x.name), val)
+
+        if isinstance(agg_fn, ds.count_cat):
```

Comment:
Hopefully a good place to add this.

Reply:
Placement looks fine to me. We also need this for `ds.by`:
```suggestion
        if isinstance(agg_fn, (ds.count_cat, ds.by)):
```

We could parameterize the test.